app/assets/javascripts/project_find_file.js.coffee

```diff
@@ -0,0 +1,109 @@
+class @ProjectFindFile
+  constructor: (@options)->
+    @filePaths = {}
+    @element = ".file-finder-input"
+    @row = "<tr class='tree-item'><td class='tree-item-file-name'><i class='fa fa-file-text-o fa-fw'></i><span class='str-truncated'><a href=%blobItemUrl>%filePath</a></span></td></tr>"
+
+    # init key event(shortcut, find file)
+    @initKeyEvent()
+
+    # focus text input box
+    $(@element).focus()
+
+    # load file list
+    @load(@options.url)
+
+  # init key event
+  initKeyEvent: ->
+    # unbind and bind keyup/keydown event at text input box
+    $(document).off "keyup"
+    $(document).off "keyup", @element
+    $(document).off "keydown", "body"
+
+    # $(document).on "keyup", @goBack
+    $(document).on "keyup", (event) =>
+      if event.keyCode == 27
+        event.preventdefault
+        location.href = @options.treeUrl
+      return false
+
+    $(document).on "keyup", @element, (event) =>
+      keyCode = event.keyCode if event
+      # up down right left arrow(37~40) esc(27) caps(20) ctrl(17) alt(18) shift(16) enter(13) cmd(91)
+      if (keyCode < 37 || keyCode > 40) && keyCode != 20 && keyCode != 17 && keyCode != 18 && keyCode != 16 && keyCode != 13 && keyCode != 91
+        @findFile($(@element).val())
+        @selecteRow $("tr.tree-item")
+
+    $(document).on "keydown", "body", (event) =>
+      keyCode = event.keyCode if event
+      if $("#search:focus").length > 0 && (keyCode == 38 || keyCode == 40 || keyCode == 13)
+        return false
+
+      # 38 up 40 down
+      if keyCode == 38 || keyCode == 40
+          @selecteRow $("tr.tree-item"), $('.tree-item.selected'), keyCode
+          return false
+      else if keyCode is 13
+        path = $('.tree-item.selected .tree-item-file-name a').attr('href')
+        if path then location.href = path
+
+  # find file
+  findFile: (searchTxt) ->
+      result = if searchTxt then fuzzaldrinPlus.filter(@filePaths, searchTxt) else @filePaths
+      @renderList result, searchTxt
+
+  # select row and focus
+  selecteRow: (rows, seletedRow, which) ->
+    if rows && rows.length > 0
+      if seletedRow && seletedRow.length > 0
+        next = null
+        # 38 up 40 down
+        if which is 38
+          next = seletedRow.prev()
+        else
+          next = seletedRow.next()
+
+        if next.length > 0
+          seletedRow.removeClass "selected"
+          seletedRow = next
+
+      else
+        seletedRow = rows.eq(0)
+      seletedRow.addClass("selected").focus()
+
+  # files pathes load
+  load: (url) ->
+    $.ajax
+      url: url
+      method: "get"
+      dataType: "json"
+      success: $.proxy((data) ->
+        $(".loading").hide()
+        @filePaths = data
+        @findFile($(@element).val())
+        @selecteRow $("tr.tree-item")
+      , this)
+
+  # render result
+  renderList: (filePaths, searchTxt) ->
+    $(".tree-table > tbody").empty()
+    for filePath, i in filePaths
+      break if i > 20

```

Comment:
This means we break when `i == 21`, which means we had 0..20 before this: 21 renders :) We need only 20.